lib/types/row.js

```diff
@@ -13,17 +37,24 @@ function Row(columns) {
  */
 Row.prototype.get = function (columnName) {
   if (typeof columnName === 'number') {
-    if (this.__columns && this.__columns[columnName]) {
-      columnName = this.__columns[columnName].name;
-    }
-    else {
-      throw new Error('Column not found');
-    }
+    return this._getValueFunc(columnName)();
```

Comment:
Yes, it's a trade-off, lazy value decoding at row level allow us to better tune memory management, similar to java's `ArrayBackedRow` and C# `Row`. It will enable us to use the same network buffer to be used as row backing without the need to copy buffers as we are used to do.